pkg/cmd/create/sampleapp/exec.go

```diff
@@ -0,0 +1,164 @@
+// Copyright Contributors to the Open Cluster Management project
+
+package sampleapp
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+
+	"github.com/spf13/cobra"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	clusterclientset "open-cluster-management.io/api/client/cluster/clientset/versioned"
+	"open-cluster-management.io/clusteradm/pkg/cmd/create/sampleapp/scenario"
+	"open-cluster-management.io/clusteradm/pkg/helpers"
+	"open-cluster-management.io/clusteradm/pkg/helpers/apply"
+)
+
+const (
+	placement            = "sampleapp"
+	defaultSampleAppName = "sampleapp"
+	pathToAppManifests   = "scenario/sampleapp"
+)
+
+func (o *Options) complete(cmd *cobra.Command, args []string) (err error) {
+
+	if len(args) > 1 {
+		return fmt.Errorf("only one sample app name can be specified")
+	}
+
+	if len(args) == 0 {
+		o.SampleAppName = defaultSampleAppName
+	} else {
+		o.SampleAppName = args[0]
+	}
+
+	return nil
+}
+
+func (o *Options) Validate() (err error) {
+
+	return nil
+}
+
+func (o *Options) Run() (err error) {
+
+	restConfig, err := o.ClusteradmFlags.KubectlFactory.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+
+	clusterClient, err := clusterclientset.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+
+	kubeClient, apiExtensionsClient, dynamicClient, err := helpers.GetClients(o.ClusteradmFlags.KubectlFactory)
+	if err != nil {
+		return err
+	}
+
+	return o.runWithClient(clusterClient, kubeClient, apiExtensionsClient, dynamicClient, o.ClusteradmFlags.DryRun)
+}
+
+func (o *Options) runWithClient(clusterClient clusterclientset.Interface,
+	kubeClient kubernetes.Interface,
+	apiExtensionsClient apiextensionsclient.Interface,
+	dynamicClient dynamic.Interface,
+	dryRun bool) error {
+
+	// Label all managed clusters with clusterset and placement labels
+	err := o.labelManagedClusters(clusterClient)
```

Comment:
could we just let user to bind the clusterset at first instead of automatically adding labels and create binding. You could pre-check the binding here and print a warning that user does not have binding in this ns yet.

Reply:
Fixed, removed automatic binding and added pre-check with warning message, Thank you